controllers/discovery/initDiscoveryUserData.js

```diff
@@ -50,6 +50,8 @@ const InitDiscoveryUserData = async (req, res) => {
     );
     let topicIds = user_topics.map((topic) => topic.topic_id);
 
+    const topicIdQuery = topicIds?.length > 0 ? `AND tp.topic_id in (:topicIds)` : '';
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The conditional check for `topicIds` length is a good addition to prevent SQL errors when `topicIds` is null or empty. However, it's important to ensure that `topicIds` is always an array to avoid potential type errors. If there's a chance that `topicIds` could be `null`, you should initialize it as an empty array.
```diff
- let topicIds = user_topics.map((topic) => topic.topic_id);
+ let topicIds = user_topics ? user_topics.map((topic) => topic.topic_id) : [];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->